### DISCUSSION TITLE
Is there a way to set interim_results in the url String?

### DISCUSSION THREAD

**Discussion by AndyWangLYN** [deepgram/community#264] [General help] [answered]
I am trying to make api call on my iOS client side using the tutoruial: https://deepgram.com/learn/ios-live-transcription

I want to enable DeepGram to return word-by-word, instead of a whole sentence. I realized this can be done using the "interim_result = true" param but I do not know if I can add this param in url call. I do not want to do this on the server side so that is why I am stuck.

Please help. Example of what I want to achieve:
let url = URL(string: "wss://api.deepgram.com/v1/listen?encoding=linear16&sample_rate=48000&channels=1&interim_results=true")!
Note I added interim_results=true at the end, but appearantly it does not work this way.

Thanks.

**Comment by jkroll-deepgram** [comment]
Hi @AndyWangLYN, you're correct that you can append that `interim_results=true` parameter to the URL. Interim results are returned approximately every 1-2 seconds, while a final transcription is returned approximately every 3-7 seconds. It sounds like you did try that parameter - what results did you see with it?

See the full feature documentation at [Interim Results](https://developers.deepgram.com/docs/interim-results
).

**Reply by AndyWangLYN** [reply]
Hi @jkroll-deepgram, thank you so much for your swift reply. I do add **interim_results=true** in my url request but it does not seem to have made any difference. The result does not show word-by-word or in a shorter time. As I said we would really want the voice input to show word-by-word instead of showing the full sentence.

The url request I write is this: **let url = URL(string: "wss://api.deepgram.com/v1/listen?encoding=linear16&sample_rate=48000&channels=1&interim_results=true")!**

Thanks!

**Reply by jkroll-deepgram** [reply, marked as answer]
Hi @AndyWangLYN, one tip based on the iOS tutorial code - it contains the line `if response.isFinal && !transcript.isEmpty {` 
By definition, interim results are not final, so you can alter the logic, for instance to `if !transcript.isEmpty {` 
This will allow you to parse the interim results as they become available!
